semgrep_output_v1.atd

```diff
@@ -686,6 +703,9 @@ type cli_output_extra <ocaml attr="deriving show"> = {
     (* EXPERIMENTAL: since: 1.11.0 *)
     ?rules_by_engine: rule_id_and_engine_kind list option;
     ?engine_requested: engine_kind option;
+
+    (* EXPERIMENTAL: since: 1.37.0 *)
+    skipped_rules: skipped_rule list;
```

Comment:
why not a ? before so new consumers can still read stuff from old producers.

Reply:
This is a mistake. Not sure what my reasoning was. I'll leave the field required in the semgrep-core output, though.

Reply:
Ah yes, I wanted to avoid a discrepancy between pysemgrep and osemgrep output. The more convenient way to define fields that default to the empty list is `~` rather than `?` (defaults to empty list vs. defaults to None) but it produces different JSON with atdpy and atdgen. The latter tries to be smart and omits the field if its value is equal to the default, causing `"skipped_rules": []` in the JSON output from Python vs. <no field> in the JSON output from OCaml.

I'll try `atdgen -j-defaults` to match the atdpy behavior. If this doesn't work, `?` should work.

Reply:
I'm happy to report `-j-defaults` works. This illustrates the complexity of atdgen. atdpy is much simpler and saner.